### PR TITLE
8339637: (tz) Update Timezone Data to 2024b

### PR DESCRIPTION
**Pull request by johnyjose30**
Backport of tz2024b changes (https://git.openjdk.org/jdk/commit/ebc17c7c8d6febd5a887309d1b7a466bcd2cc0a9)

Not clean backport as the spec changes made in mainline is not backported to jdk23u. TZdata2024 changes announced on 5th September 2024 includes changes to "EST", "MST", and "HST", i.e., changing those time zones from distinct time zones to links to other time zones. For update releases, will not make the same ZoneId changes due to current specification.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8339637](https://bugs.openjdk.org/browse/JDK-8339637) needs maintainer approval
- [x] Change requires CSR request [JDK-8342331](https://bugs.openjdk.org/browse/JDK-8342331) to be approved

### Issues
 * [JDK-8339637](https://bugs.openjdk.org/browse/JDK-8339637): (tz) Update Timezone Data to 2024b (**Enhancement** - P3 - Approved)
 * [JDK-8342331](https://bugs.openjdk.org/browse/JDK-8342331): (tz) Update Timezone Data to 2024b (**CSR**)

### Reviewers
 * [Sean Coffey](https://openjdk.org/census#coffeys) (@coffeys - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk23u.git pull/191/head:pull/191` \
`$ git checkout pull/191`

Update a local copy of the PR: \
`$ git checkout pull/191` \
`$ git pull https://git.openjdk.org/jdk23u.git pull/191/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 191`

View PR using the GUI difftool: \
`$ git pr show -t 191`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk23u/pull/191.diff">https://git.openjdk.org/jdk23u/pull/191.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk23u/pull/191#issuecomment-2418955797)

### CODE REVIEW

File: test/jdk/java/util/TimeZone/Bug6329116.java [hunk 1]
```diff
@@ -245,5 +253,12 @@ static boolean useLocalizedShortDisplayName(TimeZone tz,
 
         return false;
     }
+    static boolean checkEurope_Brussels(String expected, String got) {
+        if ((expected.equals("CET") && got.equals("MET")) || (expected.equals("CEST") && got.equals("MEST")))
+            return true;
+        else
+            return false;
+    }
+
```
Comment (naotoj) [inline test/jdk/java/util/TimeZone/Bug6329116.java:262]: I don't think this is correct. I think COMPAT's name for "MET" needs to be modified to "CET/CEST", as MET is now a link to Europe/Brussels

**Review by coffeys — COMMENTED** [review]
minor formatting comments. Looks good otherwise.

**Review by coffeys — APPROVED** [review]
LGTM

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**